src/main/scala/diplomacy/CloneModule.scala

```diff
@@ -23,7 +24,7 @@ class ClonePorts protected[shim](elts: Data*) extends Record
 class CloneModule private (model: RawModule) extends BlackBox
 {
   override def desiredName = model.name
-  val io = IO(new ClonePorts(model.getPorts.map(_.id): _*))
+  val io = IO(new ClonePorts(DataMirror.modulePorts(model).map(_._2): _*))
 }
 
 object CloneModule
```

Comment:
FWIW, this was the original prototype of [CloneModuleAsRecord](https://javadoc.io/doc/edu.berkeley.cs/chisel3_2.13/3.5.5/chisel3/experimental/package$$CloneModuleAsRecord$.html), I would just delete this code and switch to the chisel3 API.